[ci] ReadHeaderTimeout is not configured lint failures

Lint failures such as the one below are causing build failures

```
Error: healthcheckextension.go:156:11: G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)
		server: http.Server{},
		        ^
```

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -151,7 +151,3 @@ issues:
     - text: "G402:"
       linters:
         - gosec
-    # https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12566
-    - text: "G112:"
-      linters:
-        - gosec

--- a/examples/demo/server/main.go
+++ b/examples/demo/server/main.go
@@ -172,8 +172,9 @@ func main() {
 	mux := http.NewServeMux()
 	mux.Handle("/hello", otelhttp.NewHandler(handler, "/hello"))
 	server := &http.Server{
-		Addr:    ":7080",
-		Handler: mux,
+		Addr:              ":7080",
+		Handler:           mux,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	if err := server.ListenAndServe(); err != http.ErrServerClosed {
 		handleErr(err, "server failed to serve")

--- a/exporter/splunkhecexporter/client_test.go
+++ b/exporter/splunkhecexporter/client_test.go
@@ -215,7 +215,8 @@ func runMetricsExport(cfg *Config, metrics pmetric.Metrics, expectedBatchesNum i
 	rr := make(chan receivedRequest)
 	capture := CapturingData{testing: t, receivedRequest: rr, statusCode: 200, checkCompression: !cfg.DisableCompression}
 	s := &http.Server{
-		Handler: &capture,
+		Handler:           &capture,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	defer s.Close()
 	go func() {
@@ -267,7 +268,8 @@ func runTraceExport(testConfig *Config, traces ptrace.Traces, expectedBatchesNum
 	rr := make(chan receivedRequest)
 	capture := CapturingData{testing: t, receivedRequest: rr, statusCode: 200, checkCompression: !cfg.DisableCompression}
 	s := &http.Server{
-		Handler: &capture,
+		Handler:           &capture,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	defer s.Close()
 	go func() {
@@ -326,7 +328,8 @@ func runLogExport(cfg *Config, ld plog.Logs, expectedBatchesNum int, t *testing.
 	rr := make(chan receivedRequest)
 	capture := CapturingData{testing: t, receivedRequest: rr, statusCode: 200, checkCompression: !cfg.DisableCompression}
 	s := &http.Server{
-		Handler: &capture,
+		Handler:           &capture,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	defer s.Close()
 	go func() {
@@ -965,7 +968,8 @@ func TestErrorReceived(t *testing.T) {
 		panic(err)
 	}
 	s := &http.Server{
-		Handler: &capture,
+		Handler:           &capture,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	defer s.Close()
 	go func() {

--- a/internal/aws/proxy/server.go
+++ b/internal/aws/proxy/server.go
@@ -120,8 +120,9 @@ func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
 	}
 
 	return &http.Server{
-		Addr:    cfg.Endpoint,
-		Handler: handler,
+		Addr:              cfg.Endpoint,
+		Handler:           handler,
+		ReadHeaderTimeout: 20 * time.Second,
 	}, nil
 }
 

--- a/internal/aws/xray/testdata/sampleserver/sample.go
+++ b/internal/aws/xray/testdata/sampleserver/sample.go
@@ -32,8 +32,9 @@ func main() {
 		),
 	))
 	server := &http.Server{
-		Addr:    ":8000",
-		Handler: mux,
+		Addr:              ":8000",
+		Handler:           mux,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	go func() {
 		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {

--- a/receiver/cloudflarereceiver/logs.go
+++ b/receiver/cloudflarereceiver/logs.go
@@ -62,8 +62,9 @@ func newLogsReceiver(params rcvr.CreateSettings, cfg *Config, consumer consumer.
 	}
 
 	s := &http.Server{
-		TLSConfig: tlsConfig,
-		Handler:   http.HandlerFunc(recv.handleRequest),
+		TLSConfig:         tlsConfig,
+		Handler:           http.HandlerFunc(recv.handleRequest),
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 
 	recv.server = s

--- a/receiver/mongodbatlasreceiver/alerts.go
+++ b/receiver/mongodbatlasreceiver/alerts.go
@@ -133,8 +133,9 @@ func newAlertsReceiver(params rcvr.CreateSettings, baseConfig *Config, consumer 
 		return recv, nil
 	}
 	s := &http.Server{
-		TLSConfig: tlsConfig,
-		Handler:   http.HandlerFunc(recv.handleRequest),
+		TLSConfig:         tlsConfig,
+		Handler:           http.HandlerFunc(recv.handleRequest),
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	recv.server = s
 	return recv, nil

--- a/receiver/opencensusreceiver/opencensus.go
+++ b/receiver/opencensusreceiver/opencensus.go
@@ -21,6 +21,7 @@ import (
 	"net"
 	"net/http"
 	"sync"
+	"time"
 
 	agentmetricspb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1"
 	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
@@ -215,7 +216,7 @@ func (ocr *ocReceiver) httpServer() *http.Server {
 			co := cors.Options{AllowedOrigins: ocr.corsOrigins}
 			mux = cors.New(co).Handler(mux)
 		}
-		ocr.serverHTTP = &http.Server{Handler: mux}
+		ocr.serverHTTP = &http.Server{Handler: mux, ReadHeaderTimeout: 20 * time.Second}
 	}
 
 	return ocr.serverHTTP

--- a/receiver/prometheusexecreceiver/testdata/end_to_end_metrics_test/test_prometheus_exporter.go
+++ b/receiver/prometheusexecreceiver/testdata/end_to_end_metrics_test/test_prometheus_exporter.go
@@ -47,8 +47,9 @@ func server() {
 	})
 
 	server := &http.Server{
-		Addr:    fmt.Sprintf(":%v", os.Args[1]),
-		Handler: mux,
+		Addr:              fmt.Sprintf(":%v", os.Args[1]),
+		Handler:           mux,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 
 	if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {

--- a/receiver/simpleprometheusreceiver/examples/federation/prom-counter/main.go
+++ b/receiver/simpleprometheusreceiver/examples/federation/prom-counter/main.go
@@ -40,8 +40,9 @@ func initMeter() api.Meter {
 	mux := http.NewServeMux()
 	mux.Handle("/", promhttp.Handler())
 	server := &http.Server{
-		Addr:    ":8080",
-		Handler: mux,
+		Addr:              ":8080",
+		Handler:           mux,
+		ReadHeaderTimeout: 20 * time.Second,
 	}
 	go func() {
 		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {

--- a/testbed/mockdatareceivers/mockawsxrayreceiver/trace_receiver.go
+++ b/testbed/mockdatareceivers/mockawsxrayreceiver/trace_receiver.go
@@ -23,6 +23,7 @@ import (
 	"net"
 	"net/http"
 	"sync"
+	"time"
 
 	"github.com/gorilla/mux"
 	"go.opentelemetry.io/collector/component"
@@ -91,7 +92,7 @@ func (ar *MockAwsXrayReceiver) Start(_ context.Context, host component.Host) err
 	nr.HandleFunc("/TraceSegments", ar.HTTPHandlerFunc)
 
 	// create a server with the handler
-	ar.server = &http.Server{Handler: nr}
+	ar.server = &http.Server{Handler: nr, ReadHeaderTimeout: 20 * time.Second}
 
 	// run the server on a routine
 	go func() {

